text/3463-crates-io-policy-update.md

```diff
@@ -0,0 +1,233 @@
+- Start Date: 2023-07-24
+- RFC PR: [rust-lang/rfcs#3463](https://github.com/rust-lang/rfcs/pull/3463)
+
+# Summary
+[summary]: #summary
+
+The Rust community has outgrown the current crates.io policies. This RFC proposes a new "Terms of Use" policy based on prior work by PyPI, npm and GitHub.
+
+# Motivation
+[motivation]: #motivation
+
+> Why are we doing this? What use cases does it support? What is the expected outcome?
+
+crates.io has a "[Package Policies](https://crates.io/policies)" page, which describes the current, organically grown policies. A lot of support requests or questionable uses of crates.io however need explicit decisions from the crates.io team, since many cases are currently not covered by these policies. Additionally, decisions made by the team may be seen as arbitrary without written guidelines.
+
+The situation around name squatting has lately also reached unsustainable levels, and while [namespaces](https://github.com/rust-lang/rfcs/pull/3243) might help with some parts of the problem we would still need policies for name squatting namespaces then.
```

Comment:
> So what exactly is unsustainable? Is doing 'nothing' problematic? If you do nothing, what will crates.io look like in another 10 years? Will it be so much worse than now if this proposal were not accepted?

yes, we have a huge number of spam crates on crates.io already, and if we don't have any policies attempting to restrict this then in 10 years the service will be unusable because it will have grown to 90% spam and reserved names without any content.

back in the early days people were being reasonable and only taking up crate names that they would actually have content for, but once crates.io (and Rust in general) became more popular the fight for good names apparently also became more intense with people feeling that they need to reserve good names because they might use them 10 years later.

the other thing that is unsustainable is the amount of support requests we get because of people squatting large numbers of crate names and others wanting to use these names. I stopped counting the number of times I've had to reply that there is nothing we can do about spam on crates.io because our policies basically allow it...

Reply:
> we have a huge number of spam crates on crates.io already

I understand the sentiment behind this, but I wouldn't call crates spam unless they are actually... spam.

I wrote down my thoughts on this here: https://meow.social/@Hemera/111011315038751355 

A TL;DR would be that I don't think that re-classifying 'name squatting' as 'spam' is going to help with that. I consider 'name squatting' to not be a problem. And instead crates.io might be better off by reflecting how it is used.

(I do go on and suggest one such way, but that is out of scope of this comment.)

So my feedback would be to *not* include this in the current RFC and relegate it for later. As the others parts are valuable on their own.